cli/src/pcluster/models/common.py

```diff
@@ -139,12 +139,22 @@ def validate(self, raise_on_error=False):
         self.__validators = sorted(self.__validators, key=operator.attrgetter("priority"), reverse=True)
 
         # Execute validators and add results in validation_failures array
-        for attr_validator in self.__validators:
+        for validator in self.__validators:
             # Execute it by passing all the arguments
             self._validation_failures.extend(
-                attr_validator.validator_class(raise_on_error=raise_on_error).execute(**attr_validator.validator_args)
+                validator.validator_class(raise_on_error=raise_on_error).execute(**validator.validator_args)
             )
 
+        for attr, value in self.__dict__.items():
```

Comment:
I am thinking we should move this recursive validation before self.__validators execution, because the resource was  constructed from bottom to up, we may occur some cases, the validators in upper resource depends on lower resource's attribute is valid or invalid.

Reply:
Changed, good point.